llvm/unittests/Object/SYCLBINTest.cpp

```diff
@@ -133,146 +163,111 @@ TEST(SYCLBINTest, checkSYCLBINBinaryBasicIRModule) {
-  ASSERT_NE(IRMMetadataIt, IRM.Metadata->end());
-  const PropertySet &IRMMetadata = IRMMetadataIt->second;
-  EXPECT_EQ(IRMMetadata.size(), size_t{1});
-
-  // The type is currently locked to SPIR-V. This will change in the future.
-  SmallString<16> IRMMetadataTypeKey{"type"};
-  const auto &IRMMetadataTypeIt = IRMMetadata.find(IRMMetadataTypeKey);
-  ASSERT_NE(IRMMetadataTypeIt, IRMMetadata.end());
-  const PropertyValue &IRMMetadataType = IRMMetadataTypeIt->second;
-  ASSERT_EQ(IRMMetadataType.getType(), PropertyValue::Type::UINT32);
-  EXPECT_EQ(IRMMetadataType.asUint32(), uint32_t{0});
-
-  // Check that the image is the same.
-  ASSERT_EQ(Image.size(), IRM.RawIRBytes.size());
-  EXPECT_EQ(std::memcmp(Image.data(), IRM.RawIRBytes.data(), Image.size()), 0);
-}
-
-TEST(SYCLBINTest, checkSYCLBINBinaryBasicNativeDeviceCodeImage) {
-  constexpr SYCLBIN::BundleState State = SYCLBIN::BundleState::Input;
-  static constexpr char Arch[] = "some-arch";
-
-  // Create unique temporary directory for these tests
-  SmallString<128> TestDirectory;
-  {
-    ASSERT_NO_ERROR(
-        fs::createUniqueDirectory("SYCLBINTest-test", TestDirectory));
+  // Currently we have an abstract module per image.
+  ASSERT_EQ(SYCLBINObj->AbstractModules.size(), size_t{NumImages});
+
+  std::vector<decltype(Images)::const_iterator> ImageIts;
+  ImageIts.reserve(NumImages);
+  size_t ObservedIRModules = 0, ObservedNativeDeviceCodeImages = 0;
+  for (size_t I = 0; I < NumImages; ++I) {
+    const SYCLBIN::AbstractModule &AM = SYCLBINObj->AbstractModules[I];
+
+    // This metadata should be the same as in the corresponding split module, so
+    // testing should be expanded to ensure preservation.
+    ASSERT_NE(AM.Metadata.get(), nullptr);
+    EXPECT_TRUE(AM.Metadata->getPropSets().empty());
+
+    ObservedIRModules += AM.IRModules.size();
+    for (size_t J = 0; J < AM.IRModules.size(); ++J) {
+      const SYCLBIN::IRModule &IRM = AM.IRModules[J];
+      ASSERT_NE(IRM.Metadata.get(), nullptr);
+      SmallString<16> IRMMetadataKey{
+          PropertySetRegistry::SYCLBIN_IR_MODULE_METADATA};
+      const auto &IRMMetadataIt =
+          IRM.Metadata->getPropSets().find(IRMMetadataKey);
+      ASSERT_NE(IRMMetadataIt, IRM.Metadata->end());
+      const PropertySet &IRMMetadata = IRMMetadataIt->second;
+      EXPECT_EQ(IRMMetadata.size(), size_t{1});
+
+      // The type is currently locked to SPIR-V. This will change in the future.
+      SmallString<16> IRMMetadataTypeKey{"type"};
+      const auto &IRMMetadataTypeIt = IRMMetadata.find(IRMMetadataTypeKey);
+      ASSERT_NE(IRMMetadataTypeIt, IRMMetadata.end());
+      const PropertyValue &IRMMetadataType = IRMMetadataTypeIt->second;
+      ASSERT_EQ(IRMMetadataType.getType(), PropertyValue::Type::UINT32);
+      EXPECT_EQ(IRMMetadataType.asUint32(), uint32_t{0});
+
+      // Find the image that matches.
+      auto ImageMatchIt = std::find_if(
```

Comment:
`find_if` can find the element without the provided lambda using just `operator=`.

Reply:
Ah, I didn't even think of `std::vector` having an `operator=`. Good shout!